File: AccountantWeb/Models/Expense.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Accountant.Models.Enums;

namespace Accountant.Models
{
    public class Expense
    {
        public int ID { get; set; }
        
        [Required]
        [DataType(DataType.Currency)]
        public int Amount { get; set; }
        public PayOption? PayOption { get; set; }
        public DateTime DateOfPurchase { get; set; }

        [Required]
        public int ReportID { get; set; }
        public int CategoryID { get; set; }

        [Required]
        public int PurchaserID { get; set; }

        public DateTime lastModified { get; set; }
    }
}

File: AccountantAndroid/app/src/main/java/com/kunkliricsi/accountant/database/local/entities/Expense.java
package com.kunkliricsi.accountant.database.local.entities;

import com.google.gson.annotations.SerializedName;
import com.kunkliricsi.accountant.database.local.utils.Converters;

import java.util.Date;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;

@Entity(tableName = "expenses")
@TypeConverters(Converters.class)
public class Expense {

    @PrimaryKey(autoGenerate = true)
    public int id;

    @SerializedName("amount")
    @ColumnInfo(name = "amount")
    public int Amount;

    @SerializedName("payOptions")
    @ColumnInfo(name = "payoption")
    public int PayOption;

    @SerializedName("dateOfPurchase")
    @ColumnInfo(name = "dateofpurchase")
    public Date DateOfPurchase;

    @SerializedName("reportID")
    @ColumnInfo(name = "report")
    public int ReportID;

    @SerializedName("categoryID")
    @ColumnInfo(name = "category")
    public int CategoryID;

    @SerializedName("purchaserID")
    @ColumnInfo(name = "purchaser")
    public int PurchaserID;
}
